ethcore/src/state/mod.rs

```diff
@@ -486,7 +486,7 @@ impl State {
 		let mut addresses = self.cache.borrow_mut();
 		trace!("Committing cache {:?} entries", addresses.len());
 		for (address, a) in addresses.drain().filter(|&(_, ref a)| !a.is_dirty()) {
-			self.db.cache_account(address, a.account);
+			self.db.cache_account(address, a.account, a.state == AccountState::Commited);
```

Comment:
`cache_account` doesn't particularly follow as an operation i might want to do as part of `commit_cache`. one suggests loading from disk into memory and the other suggests propagating from memory back to disk.